### ISSUE TITLE
Applying salt state file with function junos.diff fails with TypeError: diff() missing 1 required positional argument: 'd_id'

### ISSUE THREAD

**Issue by rsmekala** [Juniper/salt#96]
### Description of Issue/Question

Applying salt state file with function junos.diff fails with TypeError: diff() missing 1 required positional argument: 'd_id'

```
root@olympics-salt:/srv/salt# salt "vmx1" state.apply junos_diff

vmx1:
----------
          ID: get the diff
    Function: junos.diff
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python3.4/dist-packages/salt/state.py", line 1933, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python3.4/dist-packages/salt/loader.py", line 1939, in wrapper
                  return f(*args, **kwargs)
                File "/usr/local/lib/python3.4/dist-packages/salt/states/junos.py", line 27, in wrapper
                  ret = function(*args, **kwargs)
              TypeError: diff() missing 1 required positional argument: 'd_id'
     Started: 20:09:27.477812
    Duration: 0.779 ms
     Changes:   

Summary for vmx1
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   0.779 ms
ERROR: Minions returned with non-zero exit code
root@olympics-salt:/srv/salt# cat junos_diff.sls 
get the diff:
  junos.diff

root@olympics-salt:/srv/salt# salt "vmx1" state.apply junos_diff

vmx1:
----------
          ID: get the diff
    Function: junos.diff
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python3.4/dist-packages/salt/state.py", line 1933, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python3.4/dist-packages/salt/loader.py", line 1939, in wrapper
                  return f(*args, **kwargs)
                File "/usr/local/lib/python3.4/dist-packages/salt/states/junos.py", line 27, in wrapper
                  ret = function(*args, **kwargs)
              TypeError: diff() got an unexpected keyword argument 'id'
     Started: 20:16:10.420242
    Duration: 0.716 ms
     Changes:   

Summary for vmx1
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   0.716 ms
ERROR: Minions returned with non-zero exit code
root@olympics-salt:/srv/salt# cat junos_diff.sls 
get the diff:
  junos.diff:
   - id: 0


root@olympics-salt:/srv/salt# salt "vmx1" state.apply junos_diff

vmx1:
----------
          ID: get the diff
    Function: junos.diff
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python3.4/dist-packages/salt/state.py", line 1933, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python3.4/dist-packages/salt/loader.py", line 1939, in wrapper
                  return f(*args, **kwargs)
                File "/usr/local/lib/python3.4/dist-packages/salt/states/junos.py", line 27, in wrapper
                  ret = function(*args, **kwargs)
                File "/usr/local/lib/python3.4/dist-packages/salt/states/junos.py", line 217, in diff
                  ret['changes'] = __salt__['junos.diff'](d_id)
              TypeError: diff() takes 0 positional arguments but 1 was given
     Started: 20:16:42.985934
    Duration: 2.189 ms
     Changes:   

Summary for vmx1
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   2.189 ms
ERROR: Minions returned with non-zero exit code
root@olympics-salt:/srv/salt# cat junos_diff.sls 
get the diff:
  junos.diff:
   - d_id: 0
```

**Comment by rsmekala** [comment]
Internal PR: [1428549](https://gnats.juniper.net/web/default/1428549-1)

**Comment by rsmekala** [comment]
Closed PR: [1428549](https://gnats.juniper.net/web/default/1428549-1)
